src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/StructBuilder.java

```diff
@@ -300,26 +276,26 @@ private void emitAllocatorAllocateArray() {
         append(" MemorySegment allocateArray(int len, SegmentAllocator allocator) {\n");
         incrAlign();
         indent();
-        append("return allocator.allocate(MemoryLayout.ofSequence(len, $LAYOUT()));\n");
+        append("return allocator.allocate(MemoryLayout.sequenceLayout(len, $LAYOUT()));\n");
         decrAlign();
         indent();
         append("}\n");
         decrAlign();
     }
 
-    private void emitAsRestricted() {
+    private void emitOfAddress() {
```

Comment:
I think this needs to go to, as we're kind of inferring that users want global scope. Ok if this is handled as followup patch.